drivers/source/serial_io.c

```diff
@@ -64,7 +66,20 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
 
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
+    // take mutex
+    if (sciLinMutex != NULL){
+        if (xSemaphoreTake( sciLinMutex, ( TickType_t ) UART_MUTEX_BLOCK_TIME == pdTRUE)){
```

Comment:
this should be `if (xSemaphoreTake( sciLinMutex, ( TickType_t ) UART_MUTEX_BLOCK_TIME) == pdTRUE)` ie you're trying to compare the return value of xSemaphoreTake to see if it's successful and not UART_MUTEX_BLOCK_TIME